packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -161,7 +161,13 @@ class AndroidApk extends ApplicationPackage {
       return null;
 
     final String manifestString = manifest.readAsStringSync();
-    final xml.XmlDocument document = xml.parse(manifestString);
+    xml.XmlDocument document;
+    try {
+      document = xml.parse(manifestString);
+    } on Exception {
```

Comment:
We should probably just catch `XmlParserException` here.  That object has a line/column property which we could output in the error message.